util/src/main/java/com/google/cloud/hadoop/util/BaseAbstractGoogleAsyncWriteChannel.java

```diff
@@ -196,7 +196,7 @@ private void reuploadFromCache() throws IOException {
     }
   }
 
-  private void closeInternal() {
+  protected void closeInternal() {
```

Comment:
Making internal methods protected is not right. We should avoid it. Lets either fallback to public methods or decouple from this base class altogether